From b45916445f8273b052ebf030e531166b7786f0e7 Mon Sep 17 00:00:00 2001
From: Xuefu Zhang <xuefu@uber.com>
Date: Fri, 14 Apr 2017 06:59:46 -0700
Subject: [PATCH 1074/1164] CDH-52840: HIVE-16430: Add log to show the
 cancelled query id when cancelOperation is called
 (Zhihai via Xuefu)

(cherry picked from commit 4deefcd50433c29dcf7cb4d1e422097b9cfcbca1)

Change-Id: If6942bc4084bb3b7158c71bf8c88b785eb56f0ea
---
 .../hive/service/cli/operation/SQLOperation.java   |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java b/service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java
index 34ddcdb..274443c 100644
--- a/service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java
+++ b/service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java
@@ -378,8 +378,11 @@ private synchronized void cleanup(OperationState state) throws HiveSQLException
       Future<?> backgroundHandle = getBackgroundHandle();
       if (backgroundHandle != null) {
         boolean success = backgroundHandle.cancel(true);
+        String queryId = confOverlay.get(HiveConf.ConfVars.HIVEQUERYID.varname);
         if (success) {
-          LOG.info("The running operation has been successfully interrupted.");
+          LOG.info("The running operation has been successfully interrupted: " + queryId);
+        } else if (state == OperationState.CANCELED) {
+          LOG.info("The running operation could not be cancelled, typically because it has already completed normally: " + queryId);
         }
       }
     }
@@ -406,8 +409,16 @@ private synchronized void cleanup(OperationState state) throws HiveSQLException
 
   @Override
   public void cancel(OperationState stateAfterCancel) throws HiveSQLException {
+    String queryId = null;
+    if (stateAfterCancel == OperationState.CANCELED) {
+      queryId = confOverlay.get(HiveConf.ConfVars.HIVEQUERYID.varname);
+      LOG.info("Cancelling the query execution: " + queryId);
+    }
     cleanup(stateAfterCancel);
     cleanupOperationLog();
+    if (stateAfterCancel == OperationState.CANCELED) {
+      LOG.info("Successfully cancelled the query: " + queryId);
+    }
   }
 
   @Override
-- 
1.7.9.5

